### ISSUE TITLE
Dropout

### ISSUE THREAD

**Issue by WeiFoo** [ai-se/TuneData#15]
# From Deep Neural Networks

![screen shot 2016-07-14 at 14 56 34](https://cloud.githubusercontent.com/assets/7039841/16851858/62598460-49d3-11e6-8c3d-58d9946d4594.png)

**Randomly drop units** (along with their connections) from the neural 
network during training to address overfitting!  

```
Because many of these complicated relationships will be the result of sampling noise, 
so they will exist in the training set but not in real test data even
 if it is drawn from the same distribution.
```
# Idea Proposal

 Instead of clustering data and overfit tunings to those specific nearest clustered data, **we go the opposite direction-- that's to get a more general tuning setting**.
# How
## DE frontier Evoluation❓

```
##pesudo-code

fontier = random(size=10)
folds = 3 # for row
for i in xrange(folds):
    select_train_X, select_train_Y = sample(train_X,train_Y,len(train_X)/folds)
    select_tune_X, select_tune_Y = sample(tune_X, tune_Y, len(tune_X)/folds)
    if  i == folds-1:
        tunings = DE(select_train_X, select_train_Y, select_tune_X, select_tune_Y, frontier)
        return tunings
    else:
       frontier = DE(select_train_X, select_train_Y, select_tune_X, select_tune_Y, frontier)

```
### results folds = 3, np =10,

![screen shot 2016-07-14 at 15 46 00](https://cloud.githubusercontent.com/assets/7039841/16853385/21a94264-49da-11e6-930a-b1e7ec535130.png)

![screen shot 2016-07-14 at 15 47 07](https://cloud.githubusercontent.com/assets/7039841/16853424/4175fbaa-49da-11e6-963d-5850e20bc689.png)

**Comment by timm** [comment]
to clarify what i said last time:

you wanted advice on how to take the deep learning intuition of "randomly drop nodes" to optimization

i said use distance measures to decide the abc used in DE's mutation

that's just silly since abc are expressed in terms of optjective space, not decision space

my bad
